mlflow/gateway/schemas/chat.py

```diff
@@ -1,17 +1,119 @@
-from typing import Literal, Optional
+from typing import Literal, Optional, Union
```

Comment:
```suggestion
from __future__ import annotations

from typing import Literal, Optional, Union
```

We can use `from __future__ import annotations` to simplify type annotation using [PEP604](https://peps.python.org/pep-0604/)

Reply:
@daniellok-db Filed https://github.com/mlflow/mlflow/pull/14110. Once this PR is merged, we can just run `ruff check --unsafe-fixes --fix .` to automatically transform the code.

Reply:
just tried this and it looks like it doesn't play nicely with pydantic. basemodel doesn't seem to allow the use of `|` operator